x-pack/plugins/stack_connectors/server/connector_types/openai/lib/utils.ts

```diff
@@ -76,18 +81,34 @@ export function getRequestWithStreamOption(
 export const getAxiosOptions = (
   provider: string,
   apiKey: string,
-  stream: boolean
-): { headers: Record<string, string>; responseType?: ResponseType } => {
+  stream: boolean,
+  config?: Config
+): { headers: Record<string, string>; httpsAgent?: https.Agent; responseType?: ResponseType } => {
   const responseType = stream ? { responseType: 'stream' as ResponseType } : {};
+
   switch (provider) {
     case OpenAiProviderType.OpenAi:
       return {
-        headers: { Authorization: `Bearer ${apiKey}`, ['content-type']: 'application/json' },
+        headers: { Authorization: `Bearer ${apiKey}`, 'Content-Type': 'application/json' },
         ...responseType,
       };
     case OpenAiProviderType.AzureAi:
       return {
-        headers: { ['api-key']: apiKey, ['content-type']: 'application/json' },
+        headers: { 'api-key': apiKey, 'Content-Type': 'application/json' },
+        ...responseType,
+      };
+    case OpenAiProviderType.PkiOpenAi:
+      if (!config?.certPath || !config?.keyPath) {
+        throw new Error('Certificate and key paths are required for PKI authentication');
+      }
+      const httpsAgent = new https.Agent({
+        cert: fs.readFileSync(config.certPath),
+        key: fs.readFileSync(config.keyPath),
+        rejectUnauthorized: false, // Allow self-signed certificates
```

Comment:
Have you talked to security about this?  Seems unsafe to do this - can we expose this as an additional param?

Note also that at one point we were using `rejectUnauthorized` as a config param, and were asked to change to use the "ES standard names" for this - in this case, it's `verficationMode`, which has 3 values: `full`, `certificate`, and `none`.  `certificate` allows the hostname to not match, but still checks the certificate.